rust/datafusion/src/execution/datasource.rs

```diff
@@ -39,10 +41,36 @@ pub struct CsvDataSource {
 }
 
 impl CsvDataSource {
-    pub fn new(filename: &str, schema: Arc<Schema>, batch_size: usize) -> Self {
+    pub fn new(
```

Comment:
Should we enable schema inference for csv files? I can open a JIRA for this

Reply:
@nevi-me Thanks, that would be great.